### PR TITLE
Added material_color_utilities as a dependency for flutter package.

### PR DESCRIPTION
**Pull request by darrenaustin**
In order to add the new ColorScheme generation API (as described in the [design doc](https://flutter.dev/go/colorscheme-generation)), we need to add a dependency on `material_color_utilities`. 

Breaking this out from #93463 to make it easier to review the dependency changes independent of the API changes.

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
LGTM

The updated analyzer dev dependency seems safe: https://pub.dev/packages/analyzer/changelog

File: dev/bots/allowlist.dart [hunk 1]
```diff
@@ -47,6 +47,7 @@ const Set<String> kCorePackageAllowList = <String>{
   'js',
   'logging',
   'matcher',
+  'material_color_utilities',
```
Comment (goderbauer) [inline dev/bots/allowlist.dart:50]: This has been cleared with one of the people mentioned in the comment above?

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by zanderso — APPROVED** [review]
`material_color_utilities` has no non-dev dependencies, so it will not affect the version of dependencies required for app authors.

lgtm

**Pull request merged**